### ISSUE TITLE
fetch panicked when accepting file scheme url

### ISSUE THREAD

**Issue by keroxp** [denoland/deno#4038]
```fetch.js
fetch("file:///path/to/file")
```

```
RUST_BACKTRACE=1 deno -A fetch.js
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', /rustc/5e1a799842ba6ed4a57e91f7ab9435947482f7d8/src/libcore/macros/mod.rs:15:40
stack backtrace:
   0: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
   1: core::fmt::write
   2: std::io::Write::write_fmt
   3: std::panicking::default_hook::{{closure}}
   4: std::panicking::default_hook
   5: std::panicking::rust_panic_with_hook
   6: rust_begin_unwind
   7: core::panicking::panic_fmt
   8: core::panicking::panic
   9: deno::permissions::DenoPermissions::check_net_url
  10: deno::ops::fetch::op_fetch
  11: deno::state::State::core_op::{{closure}}
  12: deno_core::ops::OpRegistry::call
  13: deno_core::isolate::Isolate::dispatch_op
  14: deno_core::bindings::send
  15: <extern "C" fn(A0) .> R as rusty_v8::support::CFnFrom<F>>::mapping::c_fn
  16: _ZN2v88internal25FunctionCallbackArguments4CallENS0_15CallHandlerInfoE
  17: _ZN2v88internal12_GLOBAL__N_119HandleApiCallHelperILb0EEENS0_11MaybeHandleINS0_6ObjectEEEPNS0_7IsolateENS0_6HandleINS0_10HeapObjectEEESA_NS8_INS0_20FunctionTemplateInfoEEENS8_IS4_EENS0_16BuiltinArgumentsE
  18: _ZN2v88internalL26Builtin_Impl_HandleApiCallENS0_16BuiltinArgumentsEPNS0_7IsolateE
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
fatal runtime error: failed to initiate panic, error 5
[1]    95204 abort      RUST_BACKTRACE=1 deno -A fetch.js
```

**Comment by keroxp** [comment]
Should file url be valid for `fetch`? 🤔

**Comment by bartlomieju** [comment]
Ref #2150. `fetch` shouldn't accept `file://` URLs - it should error out in TS without calling Rust.

**Comment by ry** [comment]
Is there a reason fetch shouldn't accept file urls?

**Comment by nayeemrmn** [comment]
@ry It's not a http server. I thought browsers generated responses around files and it would work but I just tried fetching relatively from a `file:///` URL in chrome and it says it's not supported.

```
Fetch API cannot load file:///C:/Users/Nayeem/documents/3.png. URL scheme "file" is not supported.
```

**Comment by lucacasonato** [comment]
> @ry It's not a http server. I thought browsers generated responses around files and it would work but I just tried fetching relatively from a `file:///` URL in chrome and it says it's not supported.
> 
> ```
> Fetch API cannot load file:///C:/Users/Nayeem/documents/3.png. URL scheme "file" is not supported.
> ```

It does work in Firefox though if you are on a file:// origin if I remember right. I'll test that in a second.

**Comment by kitsonk** [comment]
And discussed in #2150, Chrome allows it with a flag.  Yes, it is for security issues in browsers, but I think the expectations in Deno are there as a server workload to just follow the Deno security model and allow it.  It should respect the `allow-read` flag though versus the `allow-net` flag, IMO.

**Comment by gibsonhan** [comment]
has anyone claim this issue?

**Comment by aduh95** [comment]
Note: same happen when using `data:` scheme.

**Comment by bartlomieju** [comment]
Fixed in #4085, let's keep discussion of fetching `file://` URLs in #2150
